scm-ui/ui-webapp/src/admin/plugins/components/ExecutePendingActionModal.tsx

```diff
@@ -21,39 +21,50 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React from "react";
-import { WithTranslation, withTranslation } from "react-i18next";
+import React, {FC, useEffect, useState} from "react";
+import { useTranslation } from "react-i18next";
 import { PendingPlugins } from "@scm-manager/ui-types";
-import { apiClient, Notification } from "@scm-manager/ui-components";
-import waitForRestart from "./waitForRestart";
+import { Notification } from "@scm-manager/ui-components";
 import PluginActionModal from "./PluginActionModal";
+import { useExecutePendingPlugins } from "@scm-manager/ui-api";
+import waitForRestart from "./waitForRestart";
 
-type Props = WithTranslation & {
+type Props = {
   onClose: () => void;
   pendingPlugins: PendingPlugins;
 };
 
-class ExecutePendingActionModal extends React.Component<Props> {
-  render() {
-    const { onClose, pendingPlugins, t } = this.props;
+const ExecutePendingActionModal: FC<Props> = ({ onClose, pendingPlugins }) => {
+  const [t] = useTranslation("admin");
+  const { update, isLoading, error, isExecuted } = useExecutePendingPlugins();
+  const [restarted, setRestarted] = useState<boolean>(false);
+  const [restarting, setRestarting] = useState<boolean>(false);
+  const loading = isLoading || restarting;
 
-    return (
-      <PluginActionModal
-        description={t("plugins.modal.executePending")}
-        label={t("plugins.modal.executeAndRestart")}
-        onClose={onClose}
-        pendingPlugins={pendingPlugins}
-        execute={this.executeAndRestart}
-      >
-        <Notification type="warning">{t("plugins.modal.restartNotification")}</Notification>
-      </PluginActionModal>
-    );
-  }
+  useEffect(() => {
+    if (isExecuted) {
+      setRestarting(true);
+      waitForRestart().then(() => {
```

Comment:
Does it make sense to move the `waitForRestart` into the ui-api hooks? If we do so it would be much easier for plugin authors to handle plugin installations.

Reply:
@sdorra Would the hook then return the actual waitForRestart function ? Just trying to understand it correctly.

Reply:
Or we just wait until the server is restarted, before returning `isExecuted`

Reply:
Wouldnt we have to set the "enabled" bit of the query then so it doesnt trigger continuously even when it shouldnt be waiting ? Or did I get something wrong ?

Reply:
I actually thought about that and wasnt really sure whether that is a behaviour we want to introduce. We could also have two properties returned: isExecuted AND isRestarted. Gives the developer more to work with in terms of displaying the current state.

Reply:
Its a bit more complicated for the plugin modal, where restarting is optional.